stl/inc/algorithm

```diff
@@ -3503,24 +3503,26 @@ namespace ranges {
             return {_STD move(_First), _STD move(_Result)};
         }
 
-        template <forward_range _Rng, weakly_incrementable _Out, class _Ty, class _Pj = identity>
-            requires permutable<iterator_t<_Rng>> && indirectly_copyable<iterator_t<_Rng>, _Out>
+        template <input_range _Rng, weakly_incrementable _Out, class _Ty, class _Pj = identity>
+            requires indirectly_copyable<iterator_t<_Rng>, _Out>
                   && indirect_binary_predicate<ranges::equal_to, projected<iterator_t<_Rng>, _Pj>, const _Ty*>
         constexpr remove_copy_result<borrowed_iterator_t<_Rng>, _Out> operator()(
             _Rng&& _Range, _Out _Result, const _Ty& _Val, _Pj _Proj = {}) const {
-            auto _UResult = _Remove_copy_unchecked(
-                _Ubegin(_Range), _Uend(_Range), _Get_unwrapped_unverified(_STD move(_Result)), _Val, _Pass_fn(_Proj));
+            auto _First   = _RANGES begin(_Range);
+            auto _UResult = _Remove_copy_unchecked(_Get_unwrapped(_STD move(_First)), _Uend(_Range),
+                _Get_unwrapped_unverified(_STD move(_Result)), _Val, _Pass_fn(_Proj));

```

Comment:
Once the slew of PRs is through I would go through the other algorithms and use `_Get_unwrapped_unverified(_STD move(_Result))` consistently. No need to create a lot of noise now

Reply:
> Once the slew of PRs is through I would go through the other algorithms and use `_Get_unwrapped_unverified(_STD move(_Result))` consistently.

`_Get_unwrapped_unverified` is used only for "copy some subset of the input" algorithms like this. Algorithms for which we can determine the size of the output ahead of time as a function of input sizes will use a different unwrapping technique once we've taught the conditional sizing machinery (`_Idl_distance`) about ranges and sentinels. (There's a work item recorded for this in #39, I've been calling it out of scope for the initial implementation of Ranges due to time constraints.)